src/category_theory/sites/cocontinuous.lean

```diff
@@ -0,0 +1,220 @@
+- `category_theory.sites.Ran_is_sheaf_of_cocontinuous`: If `u : C ⥤ D` is cocontinuous, then
+`Ran u.op` (`ₚu`) as a functor `(Cᵒᵖ ⥤ A) ⥤ (Dᵒᵖ ⥤ A)` of presheaves maps sheaves to sheaves.
+
+## References
+
+* [S. MacLane, I. Moerdijk, *Sheaves in Geometry and Logic*][MM92]
+* https://stacks.math.columbia.edu/tag/00XI
+
+-/
+
+universes u
+noncomputable theory
+
+open category_theory
+open opposite
+open category_theory.presieve.family_of_elements
+open category_theory.presieve
+open category_theory.limits
+
+namespace category_theory
+namespace sites
+variables {C D A : Type u} [category.{u} C] [category.{u} D] [category.{u} A] [has_limits A]
+variables {J : grothendieck_topology C} {K : grothendieck_topology D}
+
+/--
+A functor `u : (C, J) ⥤ (D, K)` between sites is called cocontinuous if for all covering sieves
+`R` in `D`, `R.pullback u` is a covering sieve in `C`.
+-/
+structure cocontinuous (J : grothendieck_topology C) (K : grothendieck_topology D) (u : C ⥤ D) :=
+(cover_lift : ∀ {U : C} {S : sieve (u.obj U)} (hS : S ∈ K (u.obj U)), S.functor_pullback u ∈ J U)
+
+/-- A trivial example to make the inhabited linter happy. -/
+instance id_cocontinuous : inhabited (cocontinuous J J (𝟭 _)) :=
+⟨⟨λ _ S h, by { convert h, simp }⟩⟩
+
+/-!
+We will now prove that `Ran u.op` (`ₚu`) maps sheaves to sheaves if `u` is cocontinuous. This can
+be found in https://stacks.math.columbia.edu/tag/00XK. However, the proof given here uses the
+amalgamation definition of sheaves, and thus does not require that `C` or `D` has categorical
+pullbacks.
+
+For the following proof sketch, `⊆` denotes the homs on `C` and `D` as in the topological analogy.
+By definition, the presheaf `𝒢 : Dᵒᵖ ⥤ A` is a sheaf if for every sieve `S` of `U : D`, and every
+compatible family of morphisms `X ⟶ 𝒢(V)` for each `V ⊆ U : S` with a fixed source `X`,
+we can glue them into a morphism `X ⟶ 𝒢(U)`.
+
+Since the presheaf `𝒢 := (Ran u.op).obj ℱ.val` is defined via `𝒢(U) = lim_{u(V) ⊆ U} ℱ(V)`, for
+gluing the family `x` into a `X ⟶ 𝒢(U)`, it suffices to provide a `X ⟶ ℱ(Y)` for each
+`u(Y) ⊆ U`. This can be done since `{ Y' ⊆ Y : u(Y') ⊆ U ∈ S}` is a covering sieve for `Y` on
+`C` (by the cocontinuity of `u`). Thus the morphisms `X ⟶ 𝒢(u(Y')) ⟶ ℱ(Y')` can be glued into
+a morphism `X ⟶ ℱ(Y)`. This is done in `get_sections`.
+
+In `glued_limit_cone`, we verify these obtained sections are indeed compatible, and thus we obtain
+A `X ⟶ 𝒢(U)`. The remaining work is to verify that this is indeed the amalgamation and is unique.
+-/
+namespace Ran_is_sheaf_of_cocontinuous
+variables {u : C ⥤ D} (hu : cocontinuous J K u) (ℱ : Sheaf J A)
+variables {X : A} {U : D} (S : sieve U) (hS : S ∈ K U)
+variables (x : S.arrows.family_of_elements ((Ran u.op).obj ℱ.val ⋙ coyoneda.obj (op X)))
+variables (hx : x.compatible)
+
+/-- The family of morphisms `X ⟶ 𝒢(u(Y')) ⟶ ℱ(Y')` defined on `{ Y' ⊆ Y : u(Y') ⊆ U ∈ S}`. -/
+@[reducible] def pullbacked_family (Y : structured_arrow (op U) u.op) :=
```

Comment:
I'm hesitant to use `reducible` for this, unless there's a _very_ strong reason to do so, it's typically a bad sign especially for category theory things.

Reply:
I think that is a remnant from a failed attempt to make the simplifier work for me.
I have now removed it.